src/openforms/submissions/form_logic.py

```diff
@@ -81,3 +80,27 @@ def evaluate_form_logic(
     step._form_logic_evaluated = True
 
     return configuration
+
+
+def check_submission_logic(submission):
+    logic_rules = FormLogic.objects.filter(
+        form=submission.form,
+        actions__contains=[{"action": {"type": "step-not-applicable"}}],
+    )
+
+    merged_data = submission.data
+    submission_state = submission.load_execution_state()
+
+    for rule in logic_rules:
+        if jsonLogic(rule.json_logic_trigger, merged_data):
+            for action in rule.actions:
+                if action["action"]["type"] != "step-not-applicable":
+                    continue
+
+                step_to_modify_uuid = resolve(furl(action["form_step"]).pathstr).kwargs[
+                    "uuid"
+                ]
+                submission_step_to_modify = submission_state.get_submission_step(
+                    form_step_uuid=step_to_modify_uuid
+                )
```

Comment:
This is now duplicated in a couple of places - I suggest adding a method `SubmissionState.resolve_step(resource_url: str) -> SubmissionStep` which hides the implementation details